db/data_listeners.hh

```diff
@@ -155,15 +154,14 @@ public:
 
 class toppartitions_query {
     distributed<database>& _xdb;
-    sstring _ks;
-    sstring _cf;
+    std::optional<std::vector<std::tuple<sstring, sstring>>> _families;
```

Comment:
wny making a vector optional?
BTW if it was unorderd_map you could have do a quick bale-out in the implementation

Reply:
It's optional to reflect a difference between *an empty set of filters* and *no filters*. The former will match against no tables, while the latter will match against all of them

Reply:
and yeah, changing to a hashmap is a good point

Reply:
The user can omit that parameter or they can input an empty list - they mean different things and have to be handled accordingly. Once we pass that information down to the toppartitions query object and all the filtering happens, we need to be able to tell which of the described situations happened, because they give different results.
Using `std::optional` seemed like the most logical way to do it, as we're literally describing an "optional" parameter

Reply:
If I understand correctly the two option:
1. the user passes an empty container, which means they want everything
2. the user didn't pass anything, so they want nothing - so do nothing, at the topest level, just return imidiatly with an empty result

Even if you mean the logic is the other way around, I don't see why you need to implement the empty case, if a user request something that by definition returns nothing, return imidiatly with nohting.

Reply:
I removed the optionals and check for empty requests before instantiating the query, as asked